File: c#/prettygoodcuberoot.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Numerics;
using System.Threading;

namespace Kattis
{
    public class Program
    {
        public static void Main()
        {
            var s = "";
            while((s = Console.ReadLine()) != null)
            {
                var n = BigInteger.Parse(s);
                var ai = 1 + BigInteger.Parse(s)/2;
                while (BigInteger.Pow(ai, 3) > n)
                    ai = (2 * ai + n / BigInteger.Pow(ai, 2)) / 3;

                if (BigInteger.Abs(n - BigInteger.Pow(ai, 3)) > BigInteger.Abs(n - BigInteger.Pow(ai + 1, 3)))
                    ai += 1;
                Console.WriteLine(ai);
            }
        }
    }
}

File: python3/prettygoodcuberoot.py
from sys import stdin

for line in stdin:
    line = int(line)
    ai = 1+ line//3
    while (ai*ai*ai) > line:
        ai = int((2*ai + line // int(ai*ai))//3)
    if abs(line - pow(ai, 3)) > abs(line - pow(ai+1, 3)):
        ai = ai + 1
    print(ai)
